src/main/java/org/hawkular/client/android/auth/SecretStoreAuthzModule.java

```diff
@@ -0,0 +1,185 @@
+                .withIdGenerator(new IdGenerator() {
+                    @Override
+                    public String generate() {
+                        return UUID.randomUUID().toString();
+                    }
+                }).store();
+        sessionStore = (SQLStore<OAuth2AuthzSession>) DataManager.getStore(AuthEndPoints.STORE);
+        sessionStore.openSync();
+        return sessionStore;
+    }
+
+    @Override
+    public boolean isAuthorized() {
+        return true;
+    }
+
+    @Override
+    public boolean hasCredentials() {
+        OAuth2AuthzSession storedAccount = sessionStore.read(AuthEndPoints.NAME);
+        return storedAccount != null;
+    }
+
+    public void requestAccess(final Activity activity, final Callback<String> callback,
+                              final String username, final String password, final URL url) {
+
+        Exception exception;
+        String json;
+        AsyncTask<Void, Void, HeaderAndBody> task = new AsyncTask<Void, Void, HeaderAndBody>() {
+            private Exception exception;
+            String json;
+
+            @Override
+            protected HeaderAndBody doInBackground(Void... params) {
+                HeaderAndBody result;
+                try {
+                    HttpProvider provider =
+                            new HttpRestProvider(new URL(url.toString()
+                                    + AuthEndPoints.Endpoints.ACCESS));
+                    provider.setDefaultHeader("Accept", "application/json");
+                    provider.setDefaultHeader("Authorization", "Basic "
+                            + buildLoginData(username, password));
+                    result = provider.post("");
+                    Timber.e("Called with POST to: ", new String(provider.getUrl().toString()));
+                    Timber.e("Result of the POST: ", new String(result.getBody()));
+                    json = new String(result.getBody());
+                } catch (Exception e) {
+                    Timber.e(SecretStoreAuthzModule.class.getSimpleName(),
+                            "Error with Login", e);
+                    exception = e;
+                    return null;
+                }
+                return result;
+            }
+
+            @Override
+            protected void onPostExecute(HeaderAndBody headerAndBody) {
+                if (exception == null) {
+                    callback.onSuccess(new String(headerAndBody.getBody()));
+                } else {
+                    callback.onFailure(exception);
+                }
+            }
+        }.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+    }
+
+
+    @Override
+    public void requestAccess(final Activity activity, final Callback<String> callback) {
+        OAuth2AuthzSession storedAccount = sessionStore.read(AuthEndPoints.NAME);
+        final String token = storedAccount.getAccessToken();
+        try {
+            callback.onSuccess(token);
+        } catch (Exception e) {
```

Comment:
Can't know list of multiple exception which may arise. May be caused less of experience.
How to know. If net disconnect, wrong URL (that doesn't have hawkular), wrong password. And many more.